### ISSUE TITLE
Error in system creation when using `charge_from_molecules` and molecule is MOL2 loaded by OETK

### ISSUE THREAD

**Issue by j-wags** [openforcefield/openforcefield#330]
Thanks to @slochower, for finding this problem. 

Reproducing example, using 0.3.0 toolkit:

[MOL-sybyl.mol2.gz](https://github.com/openforcefield/openforcefield/files/3208985/MOL-sybyl.mol2.gz)
[CB7-sybyl.mol2.gz](https://github.com/openforcefield/openforcefield/files/3208987/CB7-sybyl.mol2.gz)

```
from openforcefield.topology import Molecule, Topology
from openforcefield.typing.engines.smirnoff import ForceField
from simtk import unit
guest = Molecule('MOL-sybyl.mol2')
host  = Molecule('CB7-sybyl.mol2')

ff = ForceField("test_forcefields/smirnoff99Frosst.offxml")

guest_system = ff.create_openmm_system(guest.to_topology(), charge_from_molecules=[guest])
host_system  = ff.create_openmm_system(host.to_topology(), charge_from_molecules=[host])
```
returns:
```
(openforcefield) jwagner@MBP-S$ python test.py 
Traceback (most recent call last):
  File "test.py", line 11, in <module>
    guest_system = ff.create_openmm_system(guest.to_topology(), charge_from_molecules=[guest])
  File "/Users/jwagner/projects/OpenForceField/openforcefield/openforcefield/typing/engines/smirnoff/forcefield.py", line 997, in create_openmm_system
    parameter_handler.create_force(system, topology, **kwargs)
  File "/Users/jwagner/projects/OpenForceField/openforcefield/openforcefield/typing/engines/smirnoff/parameters.py", line 2367, in create_force
    charges_from_charge_mol = self.assign_charge_from_molecules(temp_mol, kwargs['charge_from_molecules'])
  File "/Users/jwagner/projects/OpenForceField/openforcefield/openforcefield/typing/engines/smirnoff/parameters.py", line 2339, in assign_charge_from_molecules
    temp_mol_charges[ref_idx] = charge_mol.partial_charges[charge_idx]
  File "/Users/jwagner/miniconda3/envs/openforcefield/lib/python3.6/site-packages/simtk/unit/quantity.py", line 748, in __setitem__
    self._value[key] = value / self.unit
ValueError: assignment destination is read-only
```

We can somehow shake off the read-only nature of the partial charge array by making a deepcopy. 
Fixed code:
```
from openforcefield.topology import Molecule, Topology
from openforcefield.typing.engines.smirnoff import ForceField
from simtk import unit
guest = Molecule('MOL-sybyl.mol2')
host  = Molecule('CB7-sybyl.mol2')
import copy
guest.partial_charges = copy.deepcopy(guest.partial_charges)
host.partial_charges = copy.deepcopy(host.partial_charges)
ff = ForceField("test_forcefields/smirnoff99Frosst.offxml")

guest_system = ff.create_openmm_system(guest.to_topology(), charge_from_molecules=[guest])
host_system  = ff.create_openmm_system(host.to_topology(), charge_from_molecules=[host])
```

I suspect the problem is either in 
* `OpeneyeToolkitWrapper.from_openeye`, where we initialize storage and load partial charges
* `FrozenMolecule.partial_charges` or `TopologyMolecule.partial_charges`, setting their values as read-only
* `ToolkitAM1BCCHandler`, in being lazy about how `temp_mol_charges` is initialized

**Comment by slochower** [comment]
Thanks @j-wags for looking into this.

**Comment by SimonBoothroyd** [comment]
I'd never noticed this problem in my files - weirdly changing the example to use `Molecule.from_file` instead of just the constructor works fine!

**Comment by j-wags** [comment]
This error is no longer raised. The fix from #347 likely resolved this.
